src/http-fetcher.js

```diff
@@ -11,6 +11,20 @@ export default function httpFetcher(url, options = {}) {
         ...options.headers,
         ...headers
       }
-    }).then((response) => response.json());
+    }).then((response) => _handleResponse(response));
   };
 }
+
+function _handleResponse(response) {
+  const contentType = response.headers.get('content-type');
+
+  if (contentType.includes('application/json')) {
+    return response.json();
+  }
+
+  return _handleTextResponse(response);
+}
+
+function _handleTextResponse(response) {
+  return response.text().then((responseText) => ({data: responseText}));
+}
```

Comment:
Hey there! This is great, but I'd like to ask for a couple small changes. I feel that using the `data` key would be misleading given that the type returned is not something that's declared in any GraphQL schema. Could this key be changed to `text`? (I can add to the docs after this ships).

I'd also ask for this all to live inline, instead of passing off to a `_handleResponse` and `_handleTextResponse` method, none of these are recyclable, and really only obfuscate the flow. Could the behaviour be inlined to the fetcher?